.github/workflows/cd.yml

```diff
@@ -15,19 +16,44 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
+
       - id: auth
         uses: google-github-actions/auth@v1
         with:
-          credentials_json: ${{ secrets.GCP_SA_KEY }}
+          token_format: 'access_token'
+          workload_identity_provider: ${{ secrets.GCP_PROVIDER_NAME }}
+          service_account: ${{ secrets.GCP_SA_EMAIL }}
+
+      - name: 'Set up Cloud SDK'
+        uses: google-github-actions/setup-gcloud@v1
+
+      - id: update-secrets
+        run: |
+          gcloud config set project ${{ secrets.GCP_PROJECT_ID }}
+
+          for secret in optic-firebase-client-email optic-firebase-project-id optic-firebase-private-key
+          do
+            # delete the secret
+            gcloud secrets delete --quiet $secret || true
+
+            # re-create the secret
+            gcloud secrets create --locations=${{ secrets.GCP_CLOUDRUN_SERVICE_REGION }} --replication-policy=user-managed $secret
+          done
+
+          # update the secrets
+          echo "${{ secrets.FIREBASE_CLIENT_EMAIL }}" | tr -d '\n' | gcloud secrets versions add optic-firebase-client-email --data-file=-
+          echo "${{ secrets.FIREBASE_PROJECT_ID }}" | tr -d '\n' | gcloud secrets versions add optic-firebase-project-id --data-file=-
+          echo "${{ secrets.FIREBASE_PRIVATE_KEY_BASE64 }}" | tr -d '\n' | gcloud secrets versions add optic-firebase-private-key --data-file=-
```

Comment:
looking at this more closely, this is slightly risky. say that something breaks when we delete the secrets and before we have a chance to create them. this will cause the existing revision to fail, wouldn't it? can we see if we can come up with a more robust way to handle this? (note: I do appreciate this way of updating secrets values via CLI is not ideal)